clients/web/src/components/content-ads/content-ads.js

```diff
@@ -16,88 +42,64 @@ const adRailStyle = css`
-const ABOVE_THE_FOLD_MOBILE_MARS_ID = "pocket_mrec_1"
-const BELOW_THE_FOLD_MOBILE_MARS_ID = "pocket_mrec_2"
-const RIGHT_RAIL_1_MARS_ID = "pocket_skyscraper_1"
-const RIGHT_RAIL_2_MARS_ID = "pocket_skyscraper_2"
-
-let placements = {}
-placements[ABOVE_THE_FOLD_MARS_ID] = IABFixedSize.Billboard,
-placements[BELOW_THE_FOLD_MARS_ID] = IABFixedSize.Billboard,
-placements[ABOVE_THE_FOLD_MOBILE_MARS_ID] = IABFixedSize.MediumRectangle
-placements[BELOW_THE_FOLD_MOBILE_MARS_ID] = IABFixedSize.MediumRectangle
-placements[RIGHT_RAIL_1_MARS_ID] = IABFixedSize.Skyscaper,
-placements[RIGHT_RAIL_2_MARS_ID] = IABFixedSize.Skyscaper
+/**
+ * BillboardAboveTheFold
+ * ---
+ * Full size horizontal ad
+ * This ad collapses so there isn't a big screen jump since it sits above the content
+ */
+export function BillboardAboveTheFold({ isMobile, allowAds, resolved, targeting }) {
+  const collapse = allowAds && resolved
+  const viewport = useViewport()
+  const showMobile = viewport.width < 600 || isMobile
+  const placement = showMobile ? Placements.ABOVE_THE_FOLD_MOBILE : Placements.ABOVE_THE_FOLD
 
-export function BillboardAboveTheFold({ allowAds, targeting }) {
-  return allowAds ? (
-    <AdSlot
-      placementId={ABOVE_THE_FOLD_MARS_ID}
-      placementName={placements[ABOVE_THE_FOLD_MARS_ID]}
-      targeting={targeting}
-      instanceStyles={aboveTheFoldStyle}
-    />
-  ) : null
-}
-
-export function BillboardBelowTheFold({ allowAds, targeting }) {
-  return allowAds ? (
-    <AdSlot
-      placementId={BELOW_THE_FOLD_MARS_ID}
-      placementName={placements[BELOW_THE_FOLD_MARS_ID]}
-      targeting={targeting}
-      instanceStyles={belowTheFoldStyle}
-    />
-  ) : null
+  return (
+    <div className={`${adCollapseWrapper} ${collapse ? null : 'isCollapsed'}`}>
+      <div className="inner">
+        {allowAds ? (
+          <AdSlot placement={placement} targeting={targeting} instanceStyles={aboveTheFoldStyle} />
+        ) : null}
+      </div>
+    </div>
+  )
 }
 
-export function RectangleAboveTheFold({ allowAds, targeting }) {
-  return allowAds ? (
-    <AdSlot
-      placementId={ABOVE_THE_FOLD_MOBILE_MARS_ID}
-      placementName={placements[ABOVE_THE_FOLD_MOBILE_MARS_ID]}
-      targeting={targeting}
-      instanceStyles={aboveTheFoldStyle}
-    />
-  ) : null
-}
+/**
+ * BillboardBelowTheFold
+ * ---
+ * You are never gonna believe what this one does!
+ */
+export function BillboardBelowTheFold({ isMobile, allowAds, targeting }) {
+  const viewport = useViewport()
+  const showMobile = viewport.width < 600 || isMobile
+  const placement = showMobile ? Placements.BELOW_THE_FOLD_MOBILE : Placements.BELOW_THE_FOLD
 
-export function RectangleBelowTheFold({ allowAds, targeting }) {
   return allowAds ? (
-    <AdSlot
-      placementId={BELOW_THE_FOLD_MOBILE_MARS_ID}
-      placementName={placements[BELOW_THE_FOLD_MOBILE_MARS_ID]}
-      targeting={targeting}
-      instanceStyles={belowTheFoldStyle}
-    />
```

Comment:
Moved these two rectangles into the viewport logic... happy to put them back if need be.

Reply:
I think it's lovely, thank you for cleaning up and organizing my first-pass stuff 🫶